ament_tools/build_types/cmake.py

```diff
@@ -175,8 +175,7 @@ def _common_cmake_on_build(self, should_run_configure, context, prefix, extra_cm
         # Execute the configure step
         # (either cmake or the cmake_check_build_system make target)
         if should_run_configure:
-            cmake_args = [context.source_space]
-            cmake_args.extend(extra_cmake_args)
+            cmake_args = extra_cmake_args
```

Comment:
This will mutate the passed in parameter `extra_cmake_args` which is undesired. Please ensure to duplicate the list, e.g. using `cmake_args = list(extra_cmake_args)`.

Reply:
Oh, noob. Showing signs of pythonic rust. Fixed and tested.